tests/ndc/ndc.go

```diff
@@ -0,0 +1,2468 @@
+			}
+			historySize += s.sizeOfHistoryEvents(historyEvents.Events)
+			baseBranch = append(baseBranch, historyEvents)
+		}
+		baseVersionHistory := s.eventBatchesToVersionHistory(nil, baseBranch)
+
+		var branch1 []*historypb.History
+		branchVersionHistory1 := versionhistory.CopyVersionHistory(baseVersionHistory)
+		branchGenerator1 := baseGenerator.DeepCopy()
+		for i := 0; i < 10 && branchGenerator1.HasNextVertex(); i++ {
+			events := branchGenerator1.GetNextVertices()
+			historyEvents := &historypb.History{}
+			for _, event := range events {
+				historyEvents.Events = append(historyEvents.Events, event.GetData().(*historypb.HistoryEvent))
+			}
+			historySize += s.sizeOfHistoryEvents(historyEvents.Events)
+			branch1 = append(branch1, historyEvents)
+		}
+		branchVersionHistory1 = s.eventBatchesToVersionHistory(branchVersionHistory1, branch1)
+
+		var branch2 []*historypb.History
+		branchVersionHistory2 := versionhistory.CopyVersionHistory(baseVersionHistory)
+		branchGenerator2 := baseGenerator.DeepCopy()
+		branchGenerator2.SetVersion(branchGenerator2.GetVersion() + versionInc[0])
+		for i := 0; i < 10 && branchGenerator2.HasNextVertex(); i++ {
+			events := branchGenerator2.GetNextVertices()
+			historyEvents := &historypb.History{}
+			for _, event := range events {
+				historyEvents.Events = append(historyEvents.Events, event.GetData().(*historypb.HistoryEvent))
+			}
+			historySize += s.sizeOfHistoryEvents(historyEvents.Events)
+			branch2 = append(branch2, historyEvents)
+		}
+		branchVersionHistory2 = s.eventBatchesToVersionHistory(branchVersionHistory2, branch2)
+
+		var branch3 []*historypb.History
+		branchVersionHistory3 := versionhistory.CopyVersionHistory(baseVersionHistory)
+		branchGenerator3 := baseGenerator.DeepCopy()
+		branchGenerator3.SetVersion(branchGenerator3.GetVersion() + versionInc[1])
+		for i := 0; i < 10 && branchGenerator3.HasNextVertex(); i++ {
+			events := branchGenerator3.GetNextVertices()
+			historyEvents := &historypb.History{}
+			for _, event := range events {
+				historyEvents.Events = append(historyEvents.Events, event.GetData().(*historypb.HistoryEvent))
+			}
+			historySize += s.sizeOfHistoryEvents(historyEvents.Events)
+			branch3 = append(branch3, historyEvents)
+		}
+		branchVersionHistory3 = s.eventBatchesToVersionHistory(branchVersionHistory3, branch3)
+
+		s.applyEvents(
+			workflowID,
+			runID,
+			workflowType,
+			taskqueue,
+			baseVersionHistory,
+			baseBranch,
+			historyClient,
+		)
+		s.applyEvents(
+			workflowID,
+			runID,
+			workflowType,
+			taskqueue,
+			branchVersionHistory1,
+			branch1,
+			historyClient,
+		)
+		s.applyEvents(
+			workflowID,
+			runID,
+			workflowType,
+			taskqueue,
+			branchVersionHistory2,
+			branch2,
+			historyClient,
+		)
+		s.applyEvents(
+			workflowID,
+			runID,
+			workflowType,
+			taskqueue,
+			branchVersionHistory3,
+			branch3,
+			historyClient,
+		)
```

Comment:
what is the reason it is not a loop? Once you extract previos repetitions into a function you would be able to fill an array in a loop and then interate over the array calling applyEvents for each entry.